sql/CMakeLists.txt

```diff
@@ -211,6 +211,9 @@ ENDIF()
 IF(WIN32)
   SET(SQL_SOURCE ${SQL_SOURCE} handle_connections_win.cc winmain.cc)
 ENDIF()
+IF(MSVC)
+  ADD_COMPILE_FLAGS(item_strfunc.cc COMPILE_FLAGS "/utf-8")
```

Comment:
Not sure why it is needed, can you explain?  I think /utf-8 is fine option to use generally, and so no use *not* to use them, our sources are already prepared for that, since 1db4fc543bba0cadbdcecdd82a6584875559a667

If it is needed, it is best to add them generally, not just to a single file. Reason is that compiler would need less invocations - 
MSVC compiler cl.exe, as it is called from VS Generator, will compile multiple source files of a library with a single compiler invocation, but only for the files for which all compile options are the same.,

With that, you can add options e.g like this (also using "modern CMake" supported way)
```diff
diff --git a/cmake/os/Windows.cmake b/cmake/os/Windows.cmake
index f951eed7651..ec1029e20ac 100644
--- a/cmake/os/Windows.cmake
+++ b/cmake/os/Windows.cmake
@@ -273,6 +273,7 @@ IF(MSVC)
     STRING(APPEND CMAKE_CXX_FLAGS_RELEASE " /d2OptimizeHugeFunctions")
     STRING(APPEND CMAKE_CXX_FLAGS_RELWITHDEBINFO " /d2OptimizeHugeFunctions")
   ENDIF()
+  ADD_COMPILE_OPTIONS($<$<COMPILE_LANGUAGE:C,CXX>:/utf-8>)
 ENDIF()
```

Reply:
sure, can do that.

Reply:
@vaintroub should then the file require an appropriately modern CMake? If you're using modern CMake syntax here. Only on Windows, I mean.

Reply:
We're guaranteed to have CMake version around, with MSVC. The VS2022 comes with at least [3.24](https://learn.microsoft.com/en-us/cpp/overview/what-s-new-for-visual-cpp-in-visual-studio?view=msvc-170) (currently 3.28), VS2019 comes with [3.19](https://learn.microsoft.com/en-us/cpp/overview/what-s-new-for-cpp-2019?view=msvc-170)

older CMakes do not even know VS2022 or VS2019 generators.
So, on Windows it is guaranteed, but if needed I can add an assertion. 

However we need to update general CMake to something greater than 2.8.12 in general case, not on Windows, not on MSVC too, at some point, now that centos-7 is dead . We already are using something deprecated, and CMake is already complaining.

> CMake Deprecation Warning at libmariadb/CMakeLists.txt:5 (CMAKE_MINIMUM_REQUIRED):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

Reply:
generally, yes. but here I specifically mean to add something like
```cmake
CMAKE_MINIMUM_REQUIRED(VERSION 3.19)
```
to `Windows.cmake`.

On the other hand, if all cmakes that support VS2019+ generators are fine — then may be we don't need that line indeed.

Reply:
That line requires 3.0 iirc, external pcre requires 3.5 (this is in their CMakeLists.txt), the use of TARGET_OBJECTs of non-object library in sql/CMakeLists.txt requires iirc 3.14, so overall we might require the highest of these in the top-level CMakeLists.txt, for MSVC, but I think it is fine also to do nothing, until real problems occur.